packages/react-resizable-panels/package.json

```diff
@@ -9,11 +9,33 @@
     "url": "git+https://github.com/bvaughn/react-resizable-panels.git"
   },
   "source": "src/index.ts",
-  "main": "dist/react-resizable-panels.js",
-  "module": "dist/react-resizable-panels.module.js",
-  "types": "dist/react-resizable-panels.d.ts",
+  "main": "dist/react-resizable-panels.cjs.js",
+  "module": "dist/react-resizable-panels.esm.js",
+  "exports": {
+    ".": {
+      "types": {
+        "import": "./dist/react-resizable-panels.cjs.mjs",
+        "default": "./dist/react-resizable-panels.cjs.js"
+      },
+      "development": {
+        "module": "./dist/react-resizable-panels.development.esm.js",
+        "import": "./dist/react-resizable-panels.development.cjs.mjs",
+        "default": "./dist/react-resizable-panels.development.cjs.js"
+      },
+      "module": "./dist/react-resizable-panels.esm.js",
+      "import": "./dist/react-resizable-panels.cjs.mjs",
```

Comment:
Hey @Andarist! Looking at this tonight, to add some dev-only warnings, and wondering how I'm supposed to use this new condition. Referencing `isDevelopment` in TypeScript doesn't seem possible without directly importing one of the two files, which would seem to defeat the purpose. I'm probably overlooking something silly.

Reply:
This should work:
```ts
import { isDevelopment } from "#is-development"
```

if it doesnt - could u push out ur attempt to use it so i could take a look at the exact situation?

Reply:
I confirmed that with this patch:
```diff
diff --git a/packages/react-resizable-panels/src/PanelGroup.ts b/packages/react-resizable-panels/src/PanelGroup.ts
index 6beadfc..0c73f4f 100644
--- a/packages/react-resizable-panels/src/PanelGroup.ts
+++ b/packages/react-resizable-panels/src/PanelGroup.ts
@@ -46,6 +46,7 @@ import {
 } from "./utils/group";
 import { loadPanelLayout, savePanelGroupLayout } from "./utils/serialization";
 import { isServerRendering } from "./utils/ssr";
+import { isDevelopment } from "#is-development";
 
 const debounceMap: {
   [key: string]: (
@@ -196,7 +197,9 @@ function PanelGroupWithForwardedRef({
           0
         );
 
-        assert(total === 100, "Panel sizes must add up to 100%");
+        if (isDevelopment) {
+          assert(total === 100, "Panel sizes must add up to 100%");
+        }
 
         setSizes(sizes);
       },

```

I can only found the wrapped code in `.development.` builds:
```
> grep -R 'must add up to' packages/react-resizable-panels/dist
packages/react-resizable-panels/dist/react-resizable-panels.development.esm.js:        assert(total === 100, "Panel sizes must add up to 100%");
packages/react-resizable-panels/dist/react-resizable-panels.development.cjs.js:        assert(total === 100, "Panel sizes must add up to 100%");
```

Reply:
Ooo, neat! Thank you

Reply:
In case it's helpful to anyone else–

I tried adding this same functionality to my "suspense" library, but ran into a few troubles:

1. In order to import from the fake `"#is-development"` package, I needed to also set `moduleResolution: "bundler"` in my TS config, but doing that caused an error (it doesn't work with the `resolveJsonModule` option)
2. Removing the `resolveJsonModule` got a little further– but there were ~190 other TS errors (most suggesting "_Did you mean to set the 'moduleResolution' option to 'node'_")
3. I needed to upgrade TSC from ^4.9 to ^5.0 for the project to fix those
4. This unblocked a couple of quirks about the missing exports (seems like `export * from "path"` doesn't work if that path re-exports from another path?) but these were easy to fix
5. Now `pnpm prerelease` and `pnpm typscript` worked but `pnpm test` failed ("_Argument for '--moduleResolution' option must be: 'node', 'classic', 'node16', 'nodenext'._")

The only thing I have left to figure out is how to properly mock the fake `"#is-development"` package for testing.

Reply:
> I tried adding this same functionality to my "suspense" library, but ran into a few troubles:

Isn't frontend just fun? 🤣 

> The only thing I have left to figure out is how to properly mock the fake "#is-development" package for testing.

Do you want to test both paths (true/false)? Or is it enough to test just the development branch?

Reply:
I'm looking for a solution to test both, ideally. I'd settle for being able to only test the one though.

Reply:
Seems like everything works for building the module and running tests, but the website fails to build:

```
react-resizable-panels-website (main) ✗ p build

> packages/react-resizable-panels-website
> parcel build 'index.html'

🚨 Build failed.

@parcel/core: packages/react-resizable-panels/src/index.ts does not export 'isDevelopment'

  /Users/bvaughn/Documents/git/oss/react-resizable-panels/packages/react-resizable-panels/src/PanelGroup.ts:1:10
  > 1 | import { isDevelopment } from "#is-development";
  >   |          ^^^^^^^^^^^^
    2 | import {
    3 |   createElement,

 ELIFECYCLE  Command failed with exit code 1.
```

Reply:
Could you push what you have to a branch?

Reply:
Away from the desk for a bit but the same build error on the suspense package PR we’re talking on so that’s probably an easier starting place :)

Reply:
Hm, Parcel definitely has support for every puzzle piece here - I'm not yet sure what's wrong and since Parcel's internals are written partially in Rust I can't quite debug this in depth :S

I'll continue digging into this - there is definitely at least one Parcel bug here to be found because it crashes on existing but empty `diagnostics` internally.

If I don't come up with a fix soon-ish I will propose some alternative solution to the problem here. I think that `preconstruct dev` might be a substitute for `package.json#source` but I have to test it out with Parcel.

Reply:
Just add this to your root `package.json`:
```json
  "@parcel/resolver-default": {
    "packageExports": true
  }
```

It seems that Parcel introduced `exports` support as opt-in (weird choice if you ask me) and that impacts `imports` as well. The DX here could definitely be improved on the Parcel's side of things - at the very least they could inform the user what went wrong as the situation itself is pretty clear here. Importing a module specifier with a leading hash isn't really ambiguous as far as I know.

Reply:
Oh! That's...interesting.

Thanks for the help, @Andarist! 🙂